apps/vscode/command_client/command_client.py

```diff
@@ -368,3 +377,63 @@ def trigger_command_server_command_execution():
 class LinuxUserActions:
     def trigger_command_server_command_execution():
         actions.key("ctrl-shift-alt-p")
+
+
+@global_ctx.action_class("user")
+class GlobalUserActions:
+    def emit_pre_phrase_signal():
+        pass
+
+
+@ctx.action_class("user")
+class UserActions:
+    def emit_pre_phrase_signal():
+        get_signal_path("prePhrase").touch()
+
+
+class MissingCommunicationDir(Exception):
+    pass
+
+
+def get_signal_path(name: str) -> Path:
+    """
+    Get the path to a signal in the signal subdirectory.
+
+    Args:
+        name (str): The name of the signal
+
+    Returns:
+        Path: The signal path
+    """
+    communication_dir_path = get_communication_dir_path()
+
+    if not communication_dir_path.exists():
+        raise MissingCommunicationDir()
+
+    signal_dir = communication_dir_path / "signals"
+    signal_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
What is the point of `parents=True` when you a few lines above throw an exception if the parent/communications directory doesn't exist?

Reply:
Ha yeah thought about that. Just habit and then left it in case we decide to nest deeper down the road. I have no strong leaning tho; happy to remove it if you think that's better

Reply:
I can see the case for future proofing in it wasn't really hurt. Fine either way with me.